### ISSUE TITLE
[apex] EagerlyLoadedDescribeSObjectResult fires with deferred option

### ISSUE THREAD

**Issue by dschach** [pmd/pmd#6672]
**Affects PMD Version:** 7.25

**Rule:**

Apex EagerlyLoadedDescribeSObjectResult

Please provide the rule name and a link to the rule documentation:
<https://pmd.github.io/pmd/pmd_rules_apex_performance.html#eagerlyloadeddescribesobjectresult>

**Description:**

The rule seems to say that using DEFERRED will not cause the rule to have a violation. But I am using DEFERRED.

I am using `describeSObjects` which the rule appears to notice.

**Code Sample demonstrating the issue:**

```apex
 private static Map<String, Schema.DescribeSObjectResult> objectDescribes {
    get {
      if (objectDescribes == null) {
        objectDescribes = new Map<String, Schema.DescribeSObjectResult>();
        List<String> objectNames = new List<String>{ 'Account', 'Contact' };
        List<Schema.DescribeSobjectResult> describeResults = Schema.describeSObjects(
          objectNames,
          Schema.SObjectDescribeOptions.DEFERRED
        );
        for (Schema.DescribeSObjectResult d : describeResults) {
          objectDescribes.put(d.getName(), d);
        }
      }
      return objectDescribes;
    }
    private set;
  }
```
The important line:
```apex
Schema.describeSObjects(objectNames, Schema.SObjectDescribeOptions.DEFERRED);
```

**Expected outcome:**

PMD reports a violation at line 6, but that's wrong. That's a false positive.

**Running PMD through:** *CLI*

For reference, the rule XPath is
```xpath
//MethodCallExpression
    [
        lower-case(@MethodName) = "getdescribe" and ReferenceExpression[@SObjectType = true()]
        or lower-case(@MethodName) = "describesobjects"
    ]
    [not(VariableExpression/ReferenceExpression
            [lower-case(@Image) = ("sobjectdescribeoptions", "fielddescribeoptions")]
         )
    ]
|
//ReferenceExpression
    [$noDefault = true()]
    [lower-case(@Image) = "sobjectdescribeoptions"]
    [parent::VariableExpression[lower-case(@Image) = "default"]]
```
I have a feeling that the problem is in the rule XPath
```
lower-case(@MethodName) = "getdescribe" and ReferenceExpression[@SObjectType = true()]
        or lower-case(@MethodName) = "describesobjects"
```
which probably needs some parentheses because it is `A and B or C` which isn't a proper logical expression... but I can't fix it when using a custom XPath rule, so that isn't the cause of the false-positive.
